sdk/storage/Azure.Storage.Queues/src/QueueClient.cs

```diff
@@ -344,12 +336,18 @@ public QueueClient(Uri queueUri, TokenCredential credential, QueueClientOptions
             _uri = queueUri;
             _messagesUri = queueUri.AppendToPath(Constants.Queue.MessagesUri);
             options ??= new QueueClientOptions();
-            _pipeline = options.Build(authentication);
-            _version = options.Version;
-            _clientDiagnostics = new ClientDiagnostics(options);
-            _clientSideEncryption = QueueClientSideEncryptionOptions.CloneFrom(options._clientSideEncryptionOptions);
-            _storageSharedKeyCredential = storageSharedKeyCredential;
-            _messageEncoding = options.MessageEncoding;
+            _clientConfiguration = new QueueClientConfiguration(
+                pipeline: options.Build(authentication),
+                sharedKeyCredential: storageSharedKeyCredential,
+                clientDiagnostics: new ClientDiagnostics(options),
+                version: options.Version,
+                clientSideEncryption: QueueClientSideEncryptionOptions.CloneFrom(options._clientSideEncryptionOptions),
+                messageEncoding: options.MessageEncoding);
+
+            (QueueRestClient, MessagesRestClient, MessageIdRestClient) clients = BuildRestClients();
+            _queueRestClient = clients.Item1;
+            _messagesRestClient = clients.Item2;
+            _messageIdRestClient = clients.Item3;
```

Comment:
See https://docs.microsoft.com/en-us/dotnet/csharp/deconstruct#deconstructing-a-tuple .
Especially:
```
public static void Main()
{
    string city = "Raleigh";
    int population = 458880;
    double area = 144.8;

    (city, population, area) = QueryCityData("New York City");

    // Do something with the data.
}
```

could spare some e-ink if this would work for field assignment.

Reply:
fixed.